src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java

```diff
@@ -694,9 +688,110 @@ private String getBody(CaseData caseData) {
+                .filter(locationRefData -> checkLocation(locationRefData,
+                                                         locationLabel)).findFirst();
+
+        if (caseData.getOrderType() == OrderType.DECIDE_DAMAGES && preferredLocation.isPresent()) {
+            caseData.toBuilder().caseManagementLocation(
+                CaseLocation.builder()
+                    .region(preferredLocation.get().getRegionId())
+                    .baseLocation(preferredLocation.get().getEpimmsId()).build()
+            );
+        }
+
+        if (SPEC_CLAIM.equals(caseData.getSuperClaimType()) ||
+            (caseData.getAllocatedTrack() == AllocatedTrack.MULTI_CLAIM &&
+                SPEC_CLAIM.equals(caseData.getSuperClaimType()))) {
+            if (caseData.getApplicant1().getType() == Party.Type.INDIVIDUAL ||
+                caseData.getApplicant1().getType() == Party.Type.SOLE_TRADER) {
+                caseData.toBuilder().caseManagementLocation(
+                    CaseLocation.builder()
+                        .region(caseData.getApplicant1DQ().getRequestedCourt().getCaseLocation().getRegion())
+                        .baseLocation(caseData.getApplicant1DQ().getRequestedCourt().getCaseLocation()
+                                          .getBaseLocation()).build()
+                );
+            } else if (caseData.getApplicant1().getType() == Party.Type.COMPANY ||
+                caseData.getApplicant1().getType() == Party.Type.ORGANISATION) {
+                caseData.toBuilder().caseManagementLocation(
+                    CaseLocation.builder()
+                        .region(caseData.getRespondent1DQ().getRequestedCourt().getCaseLocation().getRegion())
+                        .baseLocation(caseData.getRespondent1DQ().getRequestedCourt().getCaseLocation()
+                                          .getBaseLocation()).build()
+                );
+            }
+        }
+
+        if (preferredLocation.isPresent()) {
+            locations.remove(preferredLocation.get());
+            locations.add(0, preferredLocation.get());
+            caseData.toBuilder().caseManagementLocation(
+                CaseLocation.builder()
+                    .region(preferredLocation.get().getRegionId())
+                    .baseLocation(preferredLocation.get().getEpimmsId()).build()
+            );
+        }
+
+        return getLocationsFromList(locations);
+    }
+
+    private Boolean checkLocation(final LocationRefData location, String locationTempLabel) {
+        return location.getEpimmsId().equals(locationTempLabel);
+    }
 
-        return locationRefDataService.getCourtLocations(authToken);
+    private List<String> getLocationsFromList(final List<LocationRefData> locations) {
+        return locations.stream()
+            .map(location -> location.getSiteName()
+                + " - " + location.getCourtAddress()
+                + " - " + location.getPostcode())
+            .collect(Collectors.toList());
     }
```

Comment:
case management location logic needs to be added in this class as per -

https://tools.hmcts.net/confluence/display/CRef/Civil+Work+Allocation+Configuration#CivilWorkAllocationConfiguration-2.2.4CaseManagementLocationrules